coder/code-server
# refactor: migrate from argon2 -> @node-rs/argon2

This migrates our hashing password implementation from using `argon2` to `@node-rs/argon2`.

Fixes https://github.com/coder/code-server/issues/4422

## How to Test

1. Wait for [CI](https://github.com/coder/code-server/actions/runs/1690093340) to finish
2. Download `release-packages` from Artifacts 
3. Unzip and then run `cd` into `release-packages`
4. Run `tar -xzf code-server*` based on your arch
5. Generate an argon2 password or use this one: "$argon2i$v=19$m=4096,t=3,p=1$cecR+o8tshB6mKzPBUG7fw$w2XFt+Ezn+CjRGTwptsjH3yj72htGLhMc3WdPs7wIOk" (equals "password")
6. edit your config.yaml and add `hashed-password: "$argon2i$v=19$m=4096,t=3,p=1$cecR+o8tshB6mKzPBUG7fw$w2XFt+Ezn+CjRGTwptsjH3yj72htGLhMc3WdPs7wIOk"`
7. Run `./code-server*/bin/code-server` 
8. Login with password: "password"
9. Everything should work

### Tested on

- [x] macos @jsjoeio 
- [x] linux @jsjoeio 
- [x] Termux @jsjoeio 
- [ ] Raspberry Pi @im-coder-lg 

## Notes

At first, I ran into some issues with tests because I was overriding `process.platform` and that caused issues with any test files that required `@node-rs/argon2`. @code-asher had a good idea to fix this: refactor `getEnvPaths()` to pass in `process.platform`. That fixed it.

### Resources

- https://crates.io/crates/argon2
- https://github.com/napi-rs/node-rs/tree/main/packages/argon2
- https://github.com/coder/code-server/issues/4422#issuecomment-1005335764
